components/locale/src/provider.rs

```diff
@@ -357,3 +370,34 @@ pub struct ScriptDirectionV1<'data> {
     #[cfg_attr(feature = "serde", serde(borrow))]
     pub ltr: ZeroVec<'data, UnvalidatedScript>,
 }
+
+/// A set of characters and strings which share a particular property value.
+///
+/// <div class="stab unstable">
+/// 🚧 This code is considered unstable; it may change at any time, in breaking or non-breaking ways,
+/// including in SemVer minor releases. While the serde representation of data structs is guaranteed
+/// to be stable, their Rust representation might not be. Use with caution.
+/// </div>
+#[icu_provider::data_struct(
+    marker(
+        ExemplarCharactersAuxiliaryV1Marker,
+        "locale/exemplarchars/auxiliary@1"
+    ),
+    marker(ExemplarCharactersIndexV1Marker, "locale/exemplarchars/index@1"),
+    marker(ExemplarCharactersMainV1Marker, "locale/exemplarchars/main@1"),
+    marker(ExemplarCharactersNumbersV1Marker, "locale/exemplarchars/numbers@1"),
+    marker(
+        ExemplarCharactersPunctuationV1Marker,
+        "locale/exemplarchars/punctuation@1"
+    )
+)]
+#[derive(Debug, Eq, PartialEq, Clone)]
+#[cfg_attr(
+    feature = "datagen", 
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_locale::provider),
+)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+pub struct ExemplarCharactersV1<'data>(
+    #[cfg_attr(feature = "serde", serde(borrow))] pub CodePointInversionListAndStringList<'data>,
```

Comment:
Note: Your new struct wraps `CodePointInversionListAndStringList` directly instead of using the enum. I think this is fine, so long as the enum in icu_properties is left alone.